token/core/zkatdlog/nogh/loaders.go

```diff
@@ -118,8 +135,12 @@ func (s *VaultPublicParamsLoader) Load() (*crypto.PublicParams, error) {
 	return pp, nil
 }
 
+// ForceFetch retrieves the public parameters from the ledger
 func (s *VaultPublicParamsLoader) ForceFetch() (*crypto.PublicParams, error) {
 	logger.Debugf("force public parameters fetch")
+	if s.PublicParamsFetcher == nil {
```

Comment:
what about adding a constructor and making VaultTokenLoader instead? We can avoid this check every time.

Reply:
For now I will remove this check.